### PR TITLE
Improvements to box vending in Requisitions

### PR DESCRIPTION
**Pull request by ThePiachu**
# About the pull request

Fixes Issue #6096

Some small improvements around the code that handles Requisitions' ammo box vendor filling logic. It now handles boxes that use handfuls (shotgun boxes) and also checks the vendor's inventory for if the boxes already exist to let the devs order them neatly as needed.

Also neatly added the various boxes to Requisitions' vendors that were auto-populated before.

# Explain why it's good for the game

It makes the vendors nicer to use and keeps the ammo boxes consistent.

# Testing Photographs and Procedure

- Check Requisitions' vendors
- Vend shotgun ammo boxes big and small to see the ratios are maitained

<details>
<summary>Screenshots & Videos</summary>

image

</details>

# Changelog
:cl: ThePiachu
qol: Shotgun ammo boxes now behave like other ammo boxes in the Requisitions' vendors.
code: Ammo box maping code now handles pre-populated boxes in vendors, letting devs order them more neatly than before.
/:cl:

### CODE REVIEW

File: code/game/machinery/vending/cm_vending.dm [hunk 1]
```diff
@@ -990,10 +990,24 @@ GLOBAL_LIST_EMPTY(vending_products)
...
+				if(tmp_item[3] == LP[3]) //We found a box we already have!
+					LP[2] = tmp_item[2] //Update box amount
+					item_found = TRUE
+					break
+			if(!item_found)
+				//We will be adding this box item at the end of the list
+				box_list += list(tmp_item)
+
+	//Putting Ammo and other boxes on the bottom of the list if they haven't been accounted for already
+	if(box_list.len > 0)
```
Comment (harryob) [inline code/game/machinery/vending/cm_vending.dm:1008]:
```suggestion
	if(length(box_list))
```

File: code/game/machinery/vending/cm_vending.dm [hunk 2]
```diff
@@ -990,10 +990,24 @@ GLOBAL_LIST_EMPTY(vending_products)
...
 
-	//Putting Ammo and other boxes on the bottom of the list as per player preferences
+	//For every item that goes into a box, check if the box is already listed in the vendor and if so, update its amount
+	var/list/box_list = list()
 	if(tmp_list.len > 0)
+		for(var/list/tmp_item as anything in tmp_list)
+			var/item_found = FALSE
+			for(var/list/LP as anything in listed_products)
+				if(tmp_item[3] == LP[3]) //We found a box we already have!
+					LP[2] = tmp_item[2] //Update box amount
```
Comment (harryob) [inline code/game/machinery/vending/cm_vending.dm:1000]:
```suggestion
			for(var/list/product as anything in listed_products)
				if(tmp_item[3] == product[3]) //We found a box we already have!
					product[2] = tmp_item[2] //Update box amount
```

**Review by harryob — CHANGES_REQUESTED** [review]

**Comment by Drulikar** [comment]
stale exempt until reviewed

**Comment by cm13-github** [comment]
This pull request has conflicts, please resolve those before we can evaluate the pull request.

**Comment by cm13-github** [comment]
Conflicts have been resolved. A maintainer will review the pull request shortly.

File: code/game/machinery/vending/cm_vending.dm [hunk 3]
```diff
@@ -990,10 +990,24 @@ GLOBAL_LIST_EMPTY(vending_products)
 		for(var/datum/item_box_pairing/IBP as anything in IMBP.item_box_pairings)
 			tmp_list += list(list(initial(IBP.box.name), floor(L[2] / IBP.items_in_box), IBP.box, VENDOR_ITEM_REGULAR))
 
-	//Putting Ammo and other boxes on the bottom of the list as per player preferences
-	if(length(tmp_list) > 0)
+	//For every item that goes into a box, check if the box is already listed in the vendor and if so, update its amount
+	var/list/box_list = list()
+	if(tmp_list.len > 0)
```
Comment (harryob) [inline code/game/machinery/vending/cm_vending.dm:995]:
```suggestion
	if(length(tmp_list))
```

**Review by harryob — CHANGES_REQUESTED** [review]

**Review by harryob — APPROVED** [review]

**Pull request merged**